lib/services/internalEngine/bootstrap-old.js

```diff
@@ -0,0 +1,331 @@
+        'roles',
+        roleId,
+        this.kuzzle.config.security.default.role)
+      );
+
+    this.kuzzle.indexCache.add(this.db.index, 'roles');
+
+    return Bluebird.all(promises);
+  }
+
+  createPluginsCollection () {
+    return this.kuzzle.indexCache.exists(this.db.index, 'plugins', false)
+      .then(exists => {
+        if (exists) {
+          return;
+        }
+
+        return this.db.updateMapping('plugins', {
+          properties: {
+            config: {
+              enabled: false
+            }
+          }
+        });
+      });
+  }
+
+  createProfilesCollection () {
+    return this.kuzzle.indexCache.exists(this.db.index, 'profiles', false)
+      .then(exists => {
+        if (exists) {
+          return;
+        }
+
+        const mapping = {
+          properties: {
+            policies: {
+              properties: {
+                roleId: {
+                  type: 'keyword'
+                }
+              }
+            }
+          }
+        };
+
+        return this.db.updateMapping('profiles', mapping)
+          .then(() => this.createDefaultProfiles());
+      });
+  }
+
+  createDefaultProfiles () {
+    const promises = ['default', 'anonymous', 'admin'].map(profileId => {
+      return this.db.createOrReplace('profiles', profileId, {
+        policies: [{roleId: profileId}]
+      });
+    });
+
+    this.kuzzle.indexCache.add(this.db.index, 'profiles');
+
+    return Bluebird.all(promises);
+  }
+
+  createUsersCollection () {
+    return this.kuzzle.indexCache.exists(this.db.index, 'users', false)
+      .then(exists => {
+        if (exists) {
+          return;
+        }
+
+        const mapping = {
+          properties: {
+            profileIds: {
+              type: 'keyword'
+            }
+          }
+        };
+
+        return this.db.updateMapping('users', mapping)
+          .then(() => this.kuzzle.indexCache.add(this.db.index, 'users'));
+      });
+  }
+
+  createValidationCollection (validationConfiguration) {
+    return this.kuzzle.indexCache.exists(this.db.index, 'validations', false)
+      .then(exists => {
+        if (exists) {
+          return;
+        }
+
+        this.kuzzle.log.info('== Creating validation collection...');
+
+        return this.db.updateMapping('validations', {
+          properties: {
+            index: {
+              type: 'keyword'
+            },
+            collection: {
+              type: 'keyword'
+            },
+            validation: {
+              enabled: false
+            }
+          }
+        });
+      })
+      .then(() => {
+        if (validationConfiguration) {
+          const promises = [];
+
+          Object.keys(validationConfiguration).forEach(indexName => {
+            Object.keys(
+              validationConfiguration[indexName]).forEach(collectionName => {
+              // createOrReplace instead of create to avoid mess on concurency
+              // TODO fix with a lock instead
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)